minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-nar-unloader/src/main/java/org/apache/nifi/minifi/nar/NarAutoUnloader.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.minifi.nar;
+
+import static java.util.Objects.requireNonNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NarAutoUnloader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NarAutoUnloader.class);
+
+    private final NarAutoUnloaderTaskFactory narAutoUnloaderTaskFactory;
+
+    private volatile NarAutoUnloaderTask narAutoUnLoaderTask;
+    private volatile boolean started = false;
+
+    public NarAutoUnloader(NarAutoUnloaderTaskFactory narAutoUnloaderTaskFactory) {
+        this.narAutoUnloaderTaskFactory = requireNonNull(narAutoUnloaderTaskFactory);
+    }
+
+    public synchronized void start() throws Exception {
+        if (!started) {
+            narAutoUnLoaderTask = narAutoUnloaderTaskFactory.createNarAutoUnloaderTask();
+
+            LOGGER.info("Starting NAR Auto-Unloader Thread for directory {}", narAutoUnLoaderTask.getAutoLoadPath());
+
+            Thread autoUnloaderThread = new Thread(narAutoUnLoaderTask);
```

Comment:
What do you think about using virtual threads here? The called service will be in waiting state for most of the cases, so it seems to be a good fit.
`Thread.ofVirtual().name("NAR Auto-Unloader").start(narAutoUnLoaderTask);`